library/extractor/src/main/java/com/google/android/exoplayer2/text/dvb/DvbTeletextSubtitleParser.java

```diff
@@ -0,0 +1,511 @@
+package com.google.android.exoplayer2.text.dvb;
+
+import android.os.Build;
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+import android.text.style.BackgroundColorSpan;
+import android.text.style.ForegroundColorSpan;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.text.Cue;
+import com.google.android.exoplayer2.util.Log;
+import com.google.android.exoplayer2.util.ParsableByteArray;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** Parses {@link Cue}s from a DVB Teletext subtitle bitstream as defined in ETSI EN 300 706 */
+/* package */ final class DvbTeletextSubtitleParser {
+
+  private static final String TAG = "DvbTeletextSubParser";
+
+  private static final int TELETEXT_HEIGHT = 576;
+  private static final int TELETEXT_COLUMNS = 40;
+  private static final int TELETEXT_ROWS = 24;
+  private static final int CHAR_HEIGHT = 21;
+
+  private static final int[] TELETEXT_COLORS = {
+      0xff000000, // Alpha Black
+      0xffff0000, // Alpha Red
+      0xff00ff00, // Alpha Green
+      0xffffff00, // Alpha Yellow
+      0xff0000ff, // Alpha Blue
+      0xffff00ff, // Alpha Magenta
+      0xff00ffff, // Alpha Cyan
+      0xffffffff, // Alpha White
+  };
```

Comment:
For lots of these you can use [Color](https://developer.android.com/reference/android/graphics/Color) constants which will be self-documenting and you can remove the comments.